webapp/packages/plugin-data-viewer/src/DataViewerPage/useDataViewerPanel.ts

```diff
@@ -0,0 +1,75 @@
+/*
+ * CloudBeaver - Cloud Database Manager
+ * Copyright (C) 2020-2024 DBeaver Corp and others
+ *
+ * Licensed under the Apache License, Version 2.0.
+ * you may not use this file except in compliance with the License.
+ */
+import { ConnectionInfoResource } from '@cloudbeaver/core-connections';
+import { useService } from '@cloudbeaver/core-di';
+import { NavNodeManagerService } from '@cloudbeaver/core-navigation-tree';
+import type { ITab } from '@cloudbeaver/plugin-navigation-tabs';
+import type { IObjectViewerTabState } from '@cloudbeaver/plugin-object-viewer';
+
+import { DataPresentationService } from '../DataPresentationService';
+import { DataViewerDataChangeConfirmationService } from '../DataViewerDataChangeConfirmationService';
+import { DataViewerTableService } from '../DataViewerTableService';
+import { DataViewerTabService } from '../DataViewerTabService';
+import { useDataViewerModel } from '../useDataViewerModel';
+
+export function useDataViewerPanel(objectViewerTab: ITab<IObjectViewerTabState>) {
+  const dataViewerTableService = useService(DataViewerTableService);
+  const navNodeManagerService = useService(NavNodeManagerService);
+  const dataViewerTabService = useService(DataViewerTabService);
+  const connectionInfoResource = useService(ConnectionInfoResource);
+  const dataPresentationService = useService(DataPresentationService);
+  const dataViewerDataChangeConfirmationService = useService(DataViewerDataChangeConfirmationService);
+
+  const model = useDataViewerModel(objectViewerTab, dataViewerTableService, async (tab, state) => {
+    const node = navNodeManagerService.getNode({
+      nodeId: tab.handlerState.objectId,
+      parentId: tab.handlerState.parentId,
+    });
+
+    if (!navNodeManagerService.isNodeHasData(node)) {
+      state.resetException();
+      return;
+    }
```

Comment:
should be handled by useDataViewerModel. Don't call `state.resetException();` in such contexts (if init not throw an exception, then it should be cleared by `useDataViewerModel`)